hudi-utilities/src/main/java/org/apache/hudi/utilities/UtilHelpers.java

```diff
@@ -235,4 +248,57 @@ public static TypedProperties readConfig(InputStream in) throws IOException {
     defaults.load(in);
     return defaults;
   }
+
+  /***
+   * call spark function get the schema through jdbc.
+   * @param options
+   * @return
+   * @throws Exception
+   */
+  public static Schema getJDBCSchema(Map<String, String> options) throws Exception {
+    scala.collection.immutable.Map<String, String> ioptions = toScalaImmutableMap(options);
+    JDBCOptions jdbcOptions = new JDBCOptions(ioptions);
+    Connection conn = JdbcUtils.createConnectionFactory(jdbcOptions).apply();
+    String url = jdbcOptions.url();
+    String table = jdbcOptions.tableOrQuery();
+    JdbcOptionsInWrite jdbcOptionsInWrite = new JdbcOptionsInWrite(ioptions);
+    boolean tableExists = JdbcUtils.tableExists(conn, jdbcOptionsInWrite);
+    if (tableExists) {
+      JdbcDialect dialect = JdbcDialects.get(url);
+      try {
+        PreparedStatement statement = conn.prepareStatement(dialect.getSchemaQuery(table));
+        try {
+          statement.setQueryTimeout(Integer.parseInt(options.get("timeout")));
+          ResultSet rs = statement.executeQuery();
+          try {
+            StructType structType;
+            if (Boolean.parseBoolean(ioptions.get("nullable").get())) {
+              structType = JdbcUtils.getSchema(rs, dialect, true);
+            } else {
+              structType = JdbcUtils.getSchema(rs, dialect, false);
+            }
+            return AvroConversionUtils.convertStructTypeToAvroSchema(structType, table, "hoodie." + table);
+          } finally {
+            rs.close();
+          }
+        } finally {
+          statement.close();
+        }
+      } finally {
+        conn.close();
+      }
```

Comment:
Would be changed to `try-with-resources`? since `try-catch-finnaly` has been optimized to `try-with-resources` in the project now.

Reply:
thank you for your advice. i wll do it.